### PR TITLE
Return Int64 from IO#write (Crystal 0.35)

### PR DESCRIPTION
**Pull request by jgillich**
See https://github.com/crystal-lang/crystal/issues/9454

Automatic formatting also did some stuff, I can get rid of that if you don't like it

### CODE REVIEW

**Comment by m-o-e** [comment]
@jgillich Thanks!

Looks good to me. I'm surprised it actually still compiles on 0.33.0 (that's what CircleCI uses).
Could you update the `.circleci/config.yml` to 0.35?  Then we can merge and release.

File: src/rucksack.cr [hunk 1]
```diff
@@ -2,17 +2,23 @@ require "openssl"
 
 class Rucksack
   class NotFound < Exception; end
+
```
Comment (m-o-e) [inline src/rucksack.cr:5]: Not a fan of these blank lines, but if that's what crystal format
wants then, oh well, probably easier to keep them than to argue
with it every time... ;)

**Comment by jgillich** [comment]
@m-o-e Done

**Comment by m-o-e** [comment]
@jgillich 
Ah!  Now there's a warning (see `make test` output):
`Warning: Deprecated OpenSSL::Digest#digest. Use `final` instead.`

Wanna give that a tackle, too? 😬

**Comment by jgillich** [comment]
Also renamed the method on Checksummer, not sure if that makes sense or not :thinking:

**Comment by m-o-e** [comment]
Yes it does.

Thanks a lot, looks great! 🙇 

Will merge, add `--error-on-warnings` to the test suite
(noticed only now that it's missing) and cut a new release. :)

**Comment by jgillich** [comment]
Ah, you should also set `crystal: ">=0.35.0"` in the `shard.yml`.

**Comment by m-o-e** [comment]
Yes, I've updated it. Release is out now, please let me know in case you
notice any problems. And thanks again for the help! :+1:

**Comment by jgillich** [comment]
Will do, thanks!

**Pull request merged**